### PR TITLE
feat(analyzer): replace silan/rgain3 with ffmpeg

### LINKED ISSUE #1221: Replace silan and rgain3 (gstreamer) with ffmpeg in analyzer
The global idea is to reduce the number of dependencies and modernize the code for the analyzer.

I propose to replace `silan` and `rgain3` (`gstreamer`) with `ffmpeg` for the silence detection analyzer and the replay gain analyzer.

It seems `silan` is hard to keep up to date across all supported distribution (maybe now it's stable ?), and `gstreamer` is not easy to work with and it ships a lot of dependencies.

[From the discussion about supported systems and dependencies](https://github.com/LibreTime/libretime/issues/1214#issuecomment-850856541), here are 2 POC for replacing the above tools with ffmpeg:

It still needs some tweaking and wrapping in a proper module.

#### Silence detection

```python
#!/usr/bin/env python3

import ffmpeg
import json


_, out = (
    ffmpeg.input("tests/test_data/44100Hz-16bit-mono.mp3")
    .audio.filter("highpass", f="1000")
    .filter("silencedetect", n=0.9, d=1)
    .output("/dev/null", format="null")
    .global_args("-hide_banner", "-nostats")
    .run(capture_stderr=True)
)

result = map(
    lambda line: line[33:],
    filter(
        lambda line: line.startswith("[silencedetect"),
        out.decode("utf-8").splitlines(),
    ),
)

silences = []
silence = {}
for line in result:
    parts = line.split()
    if parts[0].startswith("silence_start"):
        silence["start"] = parts[1]
    elif parts[0].startswith("silence_end"):
        silence["end"] = parts[1]
        silence["duration"] = parts[4]
        silences.append(silence)
        silence = {}

print(json.dumps(silences))
```

#### Replay gain

Heavily inspired from the liquidsoap replaygain extractor.

This script don't check if replaygain metadata are already present in the file, but I have this in mind.

```python
#!/usr/bin/env python3

# See https://github.com/savonet/liquidsoap/blob/main/scripts/extract-replaygain

import os
import re
import json
import ffmpeg


# Using '-filter replaygain' seems to be a cleaner way to apply the replaygain filter,
# but this is not easily doable using the ffmpeg-python library.
#
# The code below runs the following ffmpeg command:
#
# ffmpeg -vn -i <filepath> \
#   -filter_complex [0]replaygain[s0] \
#   -map [s0] -f null /dev/null \
#   -hide_banner -nostats
#

_, out = (
    ffmpeg.input("tests/test_data/44100Hz-16bit-mono.mp3", vn=None)
    .filter("replaygain")
    .output("/dev/null", format="null")
    .global_args("-hide_banner", "-nostats")
    .run(capture_stderr=True)
)

# Extract last 2 lines with the replaygain stats
out = "\n".join(out.decode("utf8").splitlines()[-2:])

replaygain = {}

track_gain_match = re.search(r".* track_gain = ([-+]?[0-9]+\.[0-9]+ dB).*", out)
if track_gain_match:
    replaygain["track_gain"] = track_gain_match.group(1)


track_peak_match = re.search(r".* track_peak = ([0-1]+\.[0-9]+).*", out)
if track_peak_match:
    replaygain["track_peak"] = track_peak_match.group(1)


print(json.dumps(replaygain))
```

I would like to have some thoughts, maybe I missed something important, and replacing the above tools isn't that easy.
Anyway, it seems like a good enhancement.

### PR DESCRIPTION
**Pull request by jooola**
This remove the annoying gstreamer dependencies and silan for ffmpeg.

More context here #1221 
Fixes #1221 

- [x] Replace rgain3 with ffmpeg
The test are not perfectly passing on the mono tracks, I guess either our previous imprementation was false about the specs or we need some more tweaks.
- [x] Replace silan with ffmpeg
Need to calculate duration, as we only have the silences

:warning: Please do not squash

### CODE REVIEW

File: .github/scripts/install-bionic.sh [hunk 1]
```diff
@@ -3,26 +3,15 @@
 # Adding repos and packages
 add-apt-repository -y ppa:libretime/libretime
 apt-get -q update
-apt-get install -y gstreamer1.0-plugins-base \
-  gstreamer1.0-plugins-good \
-  gstreamer1.0-plugins-bad \
-  gstreamer1.0-plugins-ugly \
-  libgirepository1.0-dev \
+apt-get install -y \
+  ffmpeg \
```
Comment (paddatrapper) [inline .github/scripts/install-bionic.sh:7]: These changes need to go into `installer/lib/requirements-*` too

**Comment by jooola** [comment]
This branch was almost complete and was sitting here  for a few months. I finally managed to complete the work in this branch. 

While I haven't tested this completely using vagrant, and it still needs some compatibility tweaks to match the previous `metadata` dict fields, I think this can already be reviewed.

File: analyzer/libretime_analyzer/utils.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+from loguru import logger
+
+
+def run_(*args, **kwargs) -> CompletedProcess:
+    try:
+        return run(args, stdout=PIPE, stderr=PIPE, universal_newlines=True, **kwargs)
+
+    except OSError as exception:  # executable was not found
+        cmd = args[0]
+        logger.warning(f"Failed to run: {cmd} - {exception}. Is {cmd} installed?")
```
[the remark this answers is not in the document]
Comment (jooola) [reply analyzer/libretime_analyzer/utils.py:12]: Yes it should be re-raised, good catch.

**Comment by jooola** [comment]
Having this merged is awesome !!

I am curious about the dependencies disk space reduction with this merged.

**Pull request merged**